normandy/recipes/models.py

```diff
@@ -174,3 +200,64 @@ def country(self):
     @cached_property
     def request_time(self):
         return self.request.received_at
+
+
+class ApprovalRequest(models.Model):
+    recipe = models.ForeignKey(Recipe, related_name='approval_requests')
+    created = models.DateTimeField(default=datetime.now)
+    creator = models.ForeignKey(User)
+    active = models.BooleanField(default=True)
+    approval = models.OneToOneField(Approval, null=True, related_name='approval_request')
+
+    class Meta:
+        ordering = ('-created',)
+
+    class IsNotActive(Exception):
+        pass
+
+    class ActiveRequestAlreadyExists(Exception):
+        pass
+
+    @property
+    def is_approved(self):
+        return self.approval is not None
+
+    def approve(self, user):
```

Comment:
Probably want to wrap this method in a transaction to ensure all three objects get saved together.